common/app/filters/RequestLoggingFilter.scala

```diff
@@ -1,59 +1,112 @@
-import play.api.mvc.{Result, RequestHeader, Filter}
+import play.api.mvc.{Filter, RequestHeader, Result}
 
 import scala.concurrent.Future
 import scala.util.{Failure, Random, Success}
 import conf.switches.Switches
+import net.logstash.logback.marker.LogstashMarker
+import net.logstash.logback.marker.Markers._
+import play.api.Logger
+
+import scala.collection.JavaConverters._
 
 object RequestLoggingFilter extends Filter with Logging with ExecutionContexts {
+
+  case class RequestLogger(rh: RequestHeader)(implicit internalLogger: Logger, stopWatch: StopWatch) {
+    private lazy val pseudoId = Random.nextInt(Integer.MAX_VALUE)
+    private def customFieldsMarkers(): LogstashMarker = {
+      val fields = Map(
+        "req.method" -> rh.method,
+        "req.url" -> rh.uri,
+        "req.id" -> pseudoId.toString,
+        "req.latency_ms" -> stopWatch.elapsed.toString
+      )
+      appendEntries(fields.asJava)
+    }
+
+    def info(message: String): Unit = {
+      internalLogger.logger.info(customFieldsMarkers, message)
+    }
+    def warn(message: String, error: Throwable): Unit = {
+      internalLogger.logger.warn(customFieldsMarkers, message, error)
+    }
+    def error(message: String, error: Throwable): Unit = {
+      internalLogger.logger.error(customFieldsMarkers, message, error)
+    }
+  }
+
   override def apply(next: (RequestHeader) => Future[Result])(rh: RequestHeader): Future[Result] = {
-    val stopWatch = new StopWatch
 
+    implicit val stopWatch = new StopWatch
     val result = next(rh)
-
+    val requestLogger = RequestLogger(rh)
     result onComplete {
       case Success(response) =>
         response.header.headers.get("X-Accel-Redirect") match {
           case Some(internalRedirect) =>
-            log.info(s"${rh.method} ${rh.uri} took ${stopWatch.elapsed} ms and redirected to $internalRedirect")
-
+            requestLogger.info(s"${rh.method} ${rh.uri} took ${stopWatch.elapsed} ms and redirected to $internalRedirect")
           case None =>
-            log.info(s"${rh.method} ${rh.uri} took ${stopWatch.elapsed} ms and returned ${response.header.status}")
+            requestLogger.info(s"${rh.method} ${rh.uri} took ${stopWatch.elapsed} ms and returned ${response.header.status}")
         }
 
       case Failure(error) =>
-        log.warn(s"${rh.method} ${rh.uri} failed after ${stopWatch.elapsed} ms", error)
+        requestLogger.warn(s"${rh.method} ${rh.uri} failed after ${stopWatch.elapsed} ms", error)
     }
-
     result
   }
 }
 
 object DiscussionRequestLoggingFilter extends Filter with Logging with ExecutionContexts {
+
+  case class RequestLogger(rh: RequestHeader)(implicit internalLogger: Logger, stopWatch: StopWatch) {
+    private lazy val uuid = UUID.randomUUID()
+    private def customFieldsMarkers(): LogstashMarker = {
+      val fields = Map(
+        "req.method" -> rh.method,
+        "req.url" -> rh.uri,
+        "req.uuid" -> uuid.toString,
```

Comment:
Oops forget to change this to `Random.nextInt()`. Will fix
`UUID` might have some performance implications I didn't want to introduce on every single request